jobs/JGLOBAL_EFSOI

```diff
@@ -0,0 +1,138 @@
+#!/bin/ksh
+set -x
+
+export RUN_ENVIR=${RUN_ENVIR:-"nco"}
+export PS4='$SECONDS + '
+date
+
+
+#############################
+# Source relevant config files
+#############################
+export EXPDIR=${EXPDIR:-$HOMEgfs/parm/config}
+configs="base anal eupd"
+config_path=${EXPDIR:-$NWROOT/gfs.${gfs_ver}/parm/config}
+for config in $configs; do
+    . $config_path/config.$config
+    status=$?
+    [[ $status -ne 0 ]] && exit $status
+done
+
+
+##########################################
+# Source machine runtime environment
+##########################################
+. $HOMEgfs/env/${machine}.env eupd
+status=$?
+[[ $status -ne 0 ]] && exit $status
+
+
+##############################################
+# Obtain unique process id (pid) and make temp directory
+##############################################
+export pid=${pid:-$$}
+export outid=${outid:-"LL$job"}
+export DATA=${DATA:-${DATAROOT}/${jobid:?}}
+mkdir -p $DATA
+cd $DATA
+
+
+##############################################
+# Run setpdy and initialize PDY variables
+##############################################
+export cycle="t${cyc}z"
+setpdy.sh
+. ./PDY
+
+
+##############################################
+# Determine Job Output Name on System
+##############################################
+export pgmout="OUTPUT.${pid}"
+export pgmerr=errfile
+
+
+##############################################
+# Set variables used in the exglobal script
+##############################################
+export CDATE=${CDATE:-${PDY}${cyc}}
+export CDUMP=${CDUMP:-${RUN:-"gdas"}}
+if [ $RUN_ENVIR = "nco" ]; then
+    export ROTDIR=${COMROOT:?}/$NET/$envir
+fi
+
+
+##############################################
+# Begin JOB SPECIFIC work
+##############################################
+
+GDATE=$($NDATE -$assim_freq $CDATE)
+gPDY=$(echo $GDATE | cut -c1-8)
+gcyc=$(echo $GDATE | cut -c9-10)
+
+
+export APREFIX="${CDUMP}.t${cyc}z."
+export ASUFFIX=".nemsio"
+export GPREFIX="gdas.t${gcyc}z."
+
+# LFL
+#export GSUFFIX=".nemsio"
+export GSUFFIX=".nc"
+
+
+# COMIN_GES_ENS and COMOUT_ANL_ENS are used in exglobal script
+# TO DO: Map NCO's directory into these variables
+# AFE this might have to be modified to deal with efsoi
+export COMIN_GES_ENS="$ROTDIR/enkfgdas.$gPDY/$gcyc"
+export COMOUT_ANL_ENS="$ROTDIR/enkf$CDUMP.$PDY/$cyc"
+
+
+###############################################################
+# Run relevant exglobal script
+env
+msg="HAS BEGUN on `hostname`"
+postmsg "$jlogfile" "$msg"
+$LOGSCRIPT
+
+# LFL
+#SCRgsi=/scratch1/NCEPDEV/da/Andrew.Eichmann/fv3gfs/develop-efso/global-workflow/scripts/
+SCRgsi=/scratch2/BMC/rtrr/llin/200601_EFSOI_03/global-workflow_02/scripts/
```

Comment:
Hardwired paths should not be allowed.

Reply:
Thank you for the comments.  I have removed all the hardwired paths and updated JGLOBAL_EFSOI.